app/src/actioncommands.cpp

```diff
@@ -62,6 +64,118 @@ ActionCommands::ActionCommands(QWidget* parent) : QObject(parent)
+
+    QMessageBox information(mParent);
+    information.setIcon(QMessageBox::Warning);
+    information.setText(tr("You are importing a lot of frames, beware this could take some time. Are you sure you want to proceed?"));
+    information.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+    information.setDefaultButton(QMessageBox::Yes);
+
+    MovieImporter importer(this);
+    importer.setCore(mEditor);
+
+    connect(&progressDialog, &QProgressDialog::canceled, &importer, &MovieImporter::cancel);
+
+    Status st = importer.run(filePath, mEditor->playback()->fps(), FileType::MOVIE, [&progressDialog](int prog) {
+        progressDialog.setValue(prog);
+        QApplication::processEvents();
+    }, [&progressDialog](QString progMessage) {
+        progressDialog.setLabelText(progMessage);
+    }, [&information]() {
+
+        int ret = information.exec();
+        return ret == QMessageBox::Yes;
+    });
+
+    if(st == Status::CANCELED)
+    {
+        QMessageBox::warning(mParent,
+                             st.title(),
+                             st.description(),
+                             QMessageBox::Ok,
+                             QMessageBox::Ok);
+    }
+    else if (!st.ok())
+    {
+        ErrorDialog errorDialog(st.title(), st.description(), st.details().html(), mParent);
+        errorDialog.exec();
+    }
+
+    mEditor->layers()->notifyAnimationLengthChanged();
+
+    progressDialog.setValue(100);
+    progressDialog.close();
+
+    return Status::OK;
+}
+
+Status ActionCommands::importMovieAudio()
+{
+    FileDialog fileDialog(mParent);
+    QString filePath = fileDialog.openFile(FileType::MOVIE);
+    if (filePath.isEmpty())
+    {
+        return Status::FAIL;
+    }
+
+    // Show a progress dialog, as this can take a while if you have lots of images.
+    QProgressDialog progressDialog(tr("Importing movie audio..."), tr("Abort"), 0, 100, mParent);
+    hideQuestionMark(progressDialog);
+    progressDialog.setWindowModality(Qt::WindowModal);
+    progressDialog.show();
+
+    MovieImporter importer(this);
+    importer.setCore(mEditor);
+
+    connect(&progressDialog, &QProgressDialog::canceled, &importer, &MovieImporter::cancel);
+
+    Status st = importer.run(filePath, mEditor->playback()->fps(), FileType::SOUND, [&progressDialog](int prog) {
+        progressDialog.setValue(prog);
+        QApplication::processEvents();
+    }, [](QString progressMessage) {
+        Q_UNUSED(progressMessage)
+        // Not neeeded
+    }, []() {
+        return true;
+    });
+
+    if(st == Status::CANCELED)
+    {
+        QMessageBox::warning(mParent,
+                             st.title(),
+                             st.description(),
+                             QMessageBox::Ok,
+                             QMessageBox::Ok);
+    }
+    else if (!st.ok())
+    {
+        ErrorDialog errorDialog(st.title(), st.description(), st.details().html(), mParent);
+        errorDialog.exec();
+    }
```

Comment:
This is not correct, now the user will see an empty dialog every time they cancel the operation. There's no explanation, just a window with a exclamation mark. The reason reason it was !st.ok() && st != Status::CANCEL is because it's only meant to trigger a popup if the operation fails.

Right now you'll see this if you cancel 
![image](https://user-images.githubusercontent.com/1045397/81257946-dc7b7900-9034-11ea-8ece-23f974a911bb.png)

Reply:
Yeah, I was trying to get a message to display when it was cancelled because it exceeds the frame limit so I thought I'd change the user cancelled events to SAFE, but I did not realize how many points there were where it could be cancelled by the user. And I also didn't want to use ErrorDialog for it because we don't need the extra debug information for importing something that is too long. I have updated it with a solution that should work better. Let me know what you think.